controller/cmd/destination/main.go

```diff
@@ -17,6 +17,7 @@ func main() {
 	addr := flag.String("addr", ":8089", "address to serve on")
 	metricsAddr := flag.String("metrics-addr", ":9999", "address to serve scrapable metrics on")
 	kubeConfigPath := flag.String("kubeconfig", "", "path to kube config")
+	k8sDNSZone := flag.String("kubernetes-dns-zone", "", "The DNS suffix for the local Kubernetes zone.")
```

Comment:
If i understand correctly, if a suffix was provided with a leading dot (e.g. `.foo.bar`), this will lead to some odd behavior since `strings.Split(k8sDNSZone, ".")` will end up with an empty element.

If sanitizing the input is too annoying, it would be suitable to include an example.  Or even specifying a default of `cluster.local`... (since this is a sane default)

Reply:
@ver Please take a look at the 3 more recent commits. I improved the parsing so that empty labels are rejected while a trailing "." indicating a fully-qualified name is allowed; see `splitDNSName()` and the associated unit test.

I will address "cluster.local" as the default in a separate PR.